diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/InlineClassesRule.kt

```diff
@@ -52,15 +54,19 @@ class InlineClassesRule(private val configRule: List<RulesConfig>) : Rule("inlin
 
     private fun hasValidProperties(classPsi: KtClass): Boolean {
         if (classPsi.getProperties().size == 1 && !classPsi.hasExplicitPrimaryConstructor()) {
-            return !classPsi.getProperties().first().isVar
+            return !classPsi.getProperties().single().isVar
         } else if (classPsi.getProperties().isEmpty() && classPsi.hasExplicitPrimaryConstructor()) {
             return classPsi.primaryConstructorParameters.size == 1 &&
                     !classPsi.primaryConstructorParameters.first().node.hasChildOfType(VAR_KEYWORD) &&
-                    classPsi.primaryConstructorModifierList == null
+                    classPsi.primaryConstructor?.visibilityModifierType()?.value ?: "public" == "public"
```

Comment:
```suggestion
                    classPsi.primaryConstructor?.visibilityModifierType()?.value?.let { it == "public" } ?: true
```
Maybe, something like this?